engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/NegativeValidatorForNumber.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.bv;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.constraints.Negative;
+
+/**
+ * Check that the number being validated is negative.
+ *
+ * @author Hardy Ferentschik
+ * @author Xavier Sosnovsky
+ * @author Guillaume Smet
+ */
+public class NegativeValidatorForNumber implements ConstraintValidator<Negative, Number> {
+
+	private static final short SHORT_ZERO = (short) 0;
+	private static final byte BYTE_ZERO = (byte) 0;
+
+	private boolean strict;
+
+	@Override
+	public void initialize(Negative negative) {
+		this.strict = negative.strict();
+	}
+
+	@Override
+	public boolean isValid(Number value, ConstraintValidatorContext constraintValidatorContext) {
+		// null values are valid
+		if ( value == null ) {
+			return true;
+		}
+		// handling of NaN, positive infinity and negative infinity
+		else if ( value instanceof Double ) {
```

Comment:
I reckon you won't like it, but what do you think of having dedicated validators for the individual type instaed of the huge instanceof switch here. I'd prefer that, as validator resolution is done only once, so we won't pay the price for checking the type for each validator invocation again and again.